scripts/docker/buildpack-deps/Dockerfile.ubuntu.clang.ossfuzz

```diff
@@ -91,35 +92,38 @@ RUN set -ex; \
 	rm -rf /usr/src/libprotobuf-mutator
 
 # EVMONE
+WORKDIR cd /usr/src
 RUN set -ex; \
-	cd /usr/src; \
-	git clone --branch="v0.10.0" --recurse-submodules https://github.com/ethereum/evmone.git; \
-	cd evmone; \
-	mkdir build; \
-	cd build; \
-	CXX=clang++ cmake -G Ninja -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX="/usr" ..; \
+	git clone --branch="v0.11.0" --recurse-submodules https://github.com/ethereum/evmone.git
+WORKDIR evmone/build
+RUN	CXX=clang++ cmake -G Ninja -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX="/usr" ..; \
 	ninja; \
 	ninja install/strip; \
 	rm -rf /usr/src/evmone
+WORKDIR /solidity
 
 # gmp
+WORKDIR cd /usr/src
 RUN set -ex; \
-    # Replace system installed libgmp static library
-    # with sanitized version. Do not perform apt
-    # remove because it removes mlton as well that
-    # we need for building libabicoder
-    rm -f /usr/lib/x86_64-linux-gnu/libgmp.*; \
-    rm -f /usr/include/x86_64-linux-gnu/gmp.h; \
-    cd /usr/src; \
-    wget -q 'https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz' -O gmp.tar.xz; \
-    test "$(sha256sum gmp.tar.xz)" = "fd4829912cddd12f84181c3451cc752be224643e87fac497b69edddadc49b4f2  gmp.tar.xz"; \
-    tar -xf gmp.tar.xz; \
-    cd gmp-6.2.1; \
-    ./configure --prefix=/usr --enable-static=yes; \
+	git clone --depth 1 --branch "gmp-6.2.1" https://github.com/gmp-mirror/gmp-6.2.git; \
```

Comment:
Sounds good to me.

Reply:
Nice. For now in #14767 I kept the mirror used here but I'll create a separate PR with refactors where I'll include this and other changes.

One nice thing is that this will let us drop the hard-coded hash, which is a bit problematic when the source comes from a git repo. @nikola-matic hashed the whole git checkout, which seems somewhat reproducible in CI but gave me a hard time when I was trying to update it and get the right hash on my machine. There are multiple things making it non-reproduible - the inclusion of `.git` dir, tar including dates and permissions and finally even the fact that CI has a different timezone and trying to set the time with `--mtime=1970-01-01` wasn't giving me the same timestamp. Being able to just rely on the fact that Ubuntu's source repos are trustworthy will save us this kind of headache in the future.